opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -31,35 +31,64 @@
 import os
 import sys
 import opentimelineio as otio
+from collections import Iterable
 
 lib_path = os.environ.get("OTIO_AAF_PYTHON_LIB")
 if lib_path and lib_path not in sys.path:
     sys.path += [lib_path]
 
-import aaf  # noqa (E402 module level import not at top of file)
-import aaf.storage  # noqa
-import aaf.mob  # noqa
-import aaf.define  # noqa
-import aaf.component  # noqa
-import aaf.base  # noqa
+import aaf2
+import aaf2.content
+import aaf2.mobs
+import aaf2.components
+import aaf2.core
 
 debug = False
 __names = set()
 
 
+def _get_parameter(item, parameter_name):
+    values = dict([(value.name, value) for value in item.parameters.value])
+    return values.get(parameter_name)
+
+
+def _walk_item(thing):
+    slot = thing.slot
+    if not slot:
+        return
+    segment = slot.segment
+    if isinstance(segment, aaf2.components.SourceClip):
+        yield segment
+        for item in _walk_item(segment):
+            yield item
+    # elif isinstance(segment, aaf2.components.Sequence):
+    #     clip = segment.component_at_time(thing.start_time)
+    #     if isinstance(clip, SourceClip):
+    #         yield clip
+    #         for item in clip._walk_item():
+    #             yield item
+    #     else:
+    #         raise NotImplementedError("Sequence returned %s not implemented" %  str(type(segment)))
+    elif isinstance(segment, aaf2.components.EssenceGroup):
+        yield segment
+    elif isinstance(segment, aaf2.components.Filler):
+        yield segment
+    else:
+        raise NotImplementedError("walking %s not implemented" % str(type(segment)))
```

Comment:
We're using the python3-style `format` method for string formatting in OTIO.

```suggestion
        raise NotImplementedError("walking {} not implemented".format(type(segment))
```